openssl/src/pkey.rs

```diff
@@ -524,6 +524,25 @@ impl PKey<Private> {
         ffi::d2i_AutoPrivateKey
     }
 
+    /// Deserializes a DER-formatted PKCS#8 unencrypted private key.
+    ///
+    /// This method is mainly for interoperability reasons. Encrypted keyfiles should be preferred.
+    pub fn private_key_from_pkcs8(
+        der: &[u8],
+    ) -> Result<PKey<Private>, ErrorStack>
+    {
+        unsafe {
+            ffi::init();
+            let bio = MemBioSlice::new(der)?;
+            let p8inf = cvt_p(ffi::d2i_PKCS8_PRIV_KEY_INFO_bio(
+                bio.as_ptr(),
+                ptr::null_mut(),
+            ))?;
+            cvt_p(ffi::EVP_PKCS82PKEY(p8inf))
+                .map(|p| PKey::from_ptr(p))
+        }
```

Comment:
It looks like you will need to free the PKCS8_PRIV_KEY_INFO after calling EVP_PKCS82PKEY.

Reply:
Good catch. Should be fixed now.